### PR TITLE
Add a basic pool manager for memory providers

### PR DESCRIPTION
**Pull request by kswiecicki**

### CODE REVIEW

File: source/common/ur_pool_manager.hpp [hunk 1]
```diff
@@ -191,6 +206,357 @@ pool_descriptor::create(ur_usm_pool_handle_t poolHandle,
...
+struct pool_config {
+    // TODO: Config influenced by env variables
+    // dimensions[memType] = {allocMax(MB), capacity, poolSize(MB)}
+    std::map<int32_t, std::vector<int32_t>> dimensions = {
+        {UR_USM_TYPE_DEVICE, {1, 4, 256}},
+        {UR_USM_TYPE_HOST, {1, 4, 256}},
+        {UR_USM_TYPE_SHARED, {8, 4, 256}}};
+} PoolConfig;
+
+struct proxy_pool {
```
Comment (vinser52) [inline source/common/ur_pool_manager.hpp:218]: What is the responsibility of this struct?

Comment (kswiecicki) [reply source/common/ur_pool_manager.hpp:218]: It was meant to forward the alloc/dealloc calls to the memory provider and perform pooling.

Comment (vinser52) [reply source/common/ur_pool_manager.hpp:218]: So, is it an alternative to the disjoint pool? Is it just one more implementation of the heap manager?

Comment (igchor) [reply source/common/ur_pool_manager.hpp:218]: I think this implementation is no longer needed (at least for now), the PR was not just updated yet.

Comment (vinser52) [reply source/common/ur_pool_manager.hpp:218]: Got it. It was an intermediate solution while we deal with license issues.

**Comment by kswiecicki** [comment]
Closed in favor of https://github.com/oneapi-src/unified-runtime/pull/630.

**Pull request closed without merging**